File: server/nodejs/src/infrastructure/mappers/payment-details/payment-product-3209-specific-output-mapper.ts
import type { Domain } from 'onlinepayments-sdk-nodejs';
import { PaymentProduct3209SpecificOutput } from '../../../business/domain/payments/payment-details/payment-product-3209-specific-output';

type PaymentProduct3209SpecificOutputSdk = Domain.PaymentProduct3209SpecificOutput;

export const PaymentProduct3209SpecificOutputMapper = {
    fromSdkResponse: (
        response?: PaymentProduct3209SpecificOutputSdk | null,
    ): PaymentProduct3209SpecificOutput | undefined => {
        if (!response) {
            return undefined;
        }

        return {
            buyerCompliantBankMessage: response?.buyerCompliantBankMessage,
        };
    },
};

File: server/python/app/infrastructure/mappers/payment_details/payment_product_3209_specific_output_mapper.py
from typing import Optional
from app.application.domain.payments.payment_details.payment_product_3209_specific_output import PaymentProduct3209SpecificOutput
from onlinepayments.sdk.domain.payment_product3209_specific_output import PaymentProduct3209SpecificOutput as PaymentProduct3209SpecificOutputSdk

class PaymentProduct3209SpecificOutputMapper:

    @staticmethod
    def map_from_sdk_response(
        response: Optional[PaymentProduct3209SpecificOutputSdk],
    ) -> Optional[PaymentProduct3209SpecificOutput]:
        if response is None:
            return None

        dto = PaymentProduct3209SpecificOutput()
        dto.buyer_compliant_bank_message = response.buyer_compliant_bank_message

        return dto